third_party/amd/include/TritonAMDGPUToLLVM/GCNAsmFormat.h

```diff
@@ -245,6 +245,22 @@ struct GCNBuilder {
     return modArchive.back().get();
   }
 
+  // Make the operands in argArchive follow the provided \param order.
+  void reorderArgArchive(ArrayRef<Operand *> order) {
```

Comment:
The changes in GCNAsmFormat are not needed in this PR, but I think they are still useful. Happy to move them to a separate PR.